ocs_ci/deployment/hosted_cluster.py

```diff
@@ -0,0 +1,1127 @@
+            deploy_acm_hub: (bool) Deploy ACM Hub
+            deploy_metallb: (bool) Deploy MetalLB
+            download_hcp_binary: (bool) Download HCP binary
+
+        Returns:
+            str: Name of the hosted cluster
+        """
+        if not config.ENV_DATA["platform"].lower() in HCI_PROVIDER_CLIENT_PLATFORMS:
+            raise ProviderModeNotFoundException()
+
+        self.deploy_dependencies(
+            deploy_acm_hub, deploy_cnv, deploy_metallb, download_hcp_binary
+        )
+
+        return self.create_kubevirt_ocp_cluster(name=self.name)
+
+    def deploy_dependencies(
+        self, deploy_acm_hub, deploy_cnv, deploy_metallb, download_hcp_binary
+    ):
+        """
+        Deploy dependencies for hosted OCP cluster
+        Args:
+            deploy_acm_hub: bool Deploy ACM Hub
+            deploy_cnv: bool Deploy CNV
+            deploy_metallb: bool Deploy MetalLB
+            download_hcp_binary: bool Download HCP binary
+
+        """
+        initial_default_sc = helpers.get_default_storage_class()
+        logger.info(f"Initial default StorageClass: {initial_default_sc}")
+        if not initial_default_sc == constants.CEPHBLOCKPOOL_SC:
+            logger.info(
+                f"Changing the default StorageClass to {constants.CEPHBLOCKPOOL_SC}"
+            )
+            helpers.change_default_storageclass(scname=constants.CEPHBLOCKPOOL_SC)
+        if deploy_cnv:
+            self.deploy_cnv(check_cnv_ready=True)
+        if deploy_acm_hub:
+            self.deploy_acm_hub()
+        if deploy_metallb:
+            self.deploy_lb()
+        if download_hcp_binary:
+            self.update_hcp_binary()
+
+        provider_ocp_version = str(
+            get_semantic_version(get_ocp_version(), only_major_minor=True)
+        )
+        latest_released_ocp_version = str(
+            get_semantic_version(get_latest_release_version(), only_major_minor=True)
+        )
+
+        if provider_ocp_version > latest_released_ocp_version:
+            if not self.hypershift_upstream_installed():
+                try:
+                    self.disable_multicluster_engine()
+                    # avoid timelapse error "
+                    # Error: [serviceaccounts "operator" is forbidden: unable to create new content"
+                    logger.info(
+                        "Sleeping for 5 minutes after disable_multicluster_engine()"
+                    )
+                    time.sleep(5 * 60)
+                    self.install_hypershift_upstream_on_cluster()
+                except CommandFailed as e:
+                    raise AssertionError(
+                        f"Failed to install Hypershift on the cluster: {e}"
+                    )
+
+
+class HostedODF(HypershiftHostedOCP):
+    def __init__(self, name: str):
+        HyperShiftBase.__init__(self)
+        HypershiftHostedOCP.__init__(self, name)
+        self.namespace_client = constants.OPENSHIFT_STORAGE_CLIENT_NAMESPACE
```

Comment:
This value should be taken from config in case that client is installed in different namespace. It can be added to default config so that it is not defined again in other configs that need a different value. (e.g. add `client_namespace` to https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/conf/default_config.yaml#L149)

Reply:
thanks, corrected